### PR TITLE
[infra-proxy-service] User management - GET Server users list API

### PR DESCRIPTION
**Pull request by sonali523**
### :nut_and_bolt: Description: What code changed, and why?
List the users in a server

### :chains: Related Resources
https://github.com/chef/automate/issues/5554

### :+1: Definition of Done
I have added API to list the users in a server

### :athletic_shoe: How to Build and Test the Change
Steps to build:
```
rebuild components/infra-proxy-service
rebuild components/automate-gateway
```
Steps to test:

1. Ensure Chef Infra Server is running. If it is not start it with ```start_chef_server```
2. Add the chef-server test environment credentials and extract the SERVER IDs
3. Add the server admin name and admin key in the request body

Request

```curl -sSkH "api-token: $(get_admin_token)" 'https://a2-dev.test/api/v0/infra/servers/SERVER_ID/users'```

request body
```
{
    "admin_name":"ADMIN_NAME",
    "admin_key":"ADMIN_KEY"
}
```

Response

```
{
    "users": [
       {
            "username": "sonali",
            "first_name": "Sonali",
            "last_name": "Wale",
            "email": "swale@progress.com"
        },
        {
            "username": "vinay",
            "first_name": "Vinay",
            "last_name": "Sharma",
            "email": "vsharma@chef.io"
        }
    ]
}
```

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/infra-proxy-service/server/proxy.go [hunk 1]
```diff
@@ -198,3 +198,45 @@ func StructToJSON(data *structpb.Struct) (interface{}, error) {
 
 	return v, err
 }
+
+//createChefClient: Creates a client with only server details
+func (s *Server) createChefClient(ctx context.Context, serverID string, adminkey string, adminName string) (*ChefClient, error) {
```
Comment (kalroy) [inline components/infra-proxy-service/server/proxy.go:203]: Rename it as `createChefServerClient`

**Review by kalroy — CHANGES_REQUESTED** [review]

File: api/external/infra_proxy/response/users.proto [hunk 2]
```diff
@@ -12,3 +12,19 @@ message UsersListItem {
     // User username
     string username =1;
 }
+
+message ServerUsers {
+    // Users list
+    repeated ServerUserListItem users=1;
+}
+
+message ServerUserListItem {
```
[the remark this answers is not in the document]
Comment (sonali523) [reply api/external/infra_proxy/response/users.proto]: Updated the changes, please review

File: api/interservice/infra_proxy/response/users.proto [hunk 3]
```diff
@@ -13,3 +13,19 @@ message UsersListItem {
     // User username
     string username =1;
 }
+
+message ServerUsers {
+    // Users list
+    repeated ServerUserListItem users=1;
+}
+
+message ServerUserListItem {
```
[the remark this answers is not in the document]
Comment (sonali523) [reply api/interservice/infra_proxy/response/users.proto]: Did the suggested changes

File: components/infra-proxy-service/server/proxy.go (hunk 1 above)
Comment (sonali523) [reply components/infra-proxy-service/server/proxy.go]: renamed to ```createChefServerClient```

**Review by vinay033 — APPROVED** [review]

**Review by kalroy — APPROVED** [review]

**Pull request merged**